tools/gradle-plugin/src/main/kotlin/io/github/composegears/valkyrie/gradle/ValkyrieGradlePlugin.kt

```diff
@@ -0,0 +1,83 @@
+package io.github.composegears.valkyrie.gradle
+
+import com.android.build.api.dsl.CommonExtension
+import io.github.composegears.valkyrie.gradle.GenerateSvgImageVectorTask.Companion.TASK_NAME
+import java.io.File
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.SourceSet
+import org.gradle.language.base.plugins.LifecycleBasePlugin
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+import org.jetbrains.kotlin.gradle.tasks.AbstractKotlinCompile
+
+@Suppress("unused") // Registered as a Gradle plugin.
+class ValkyrieGradlePlugin : Plugin<Project> {
+    override fun apply(target: Project) = with(target) {
+        val extension = extensions.create("valkyrie", ValkyrieExtension::class.java)
+
+        val svgTask = tasks.register(TASK_NAME, GenerateSvgImageVectorTask::class.java) { task ->
+            task.group = LifecycleBasePlugin.BUILD_GROUP
+            task.description = "Converts SVG files into generated ImageVector Kotlin accessor properties"
+            task.svgFiles.setFrom(findSvgFiles())
+            task.drawableFiles.setFrom(findDrawableFiles())
+            task.packageName.set(extension.packageName.orNull ?: estimatePackageNameOrThrow())
+            task.config.set(extension.config)
+            task.outputSourceSet.set(extension.outputSourceSet.orNull ?: estimateSourceSet())
+            val defaultOutputDir = layout.buildDirectory.dir("generated/sources/valkyrie")
+            task.outputDirectory.set(extension.outputDirectory.orNull ?: defaultOutputDir.get())
+        }
+
+        afterEvaluate {
+            // Run generation immediately if we're syncing Intellij/Android Studio - helps to speed up dev cycle
+            val isIdeSyncing = System.getProperty("idea.sync.active") == "true"
+            if (extension.generateAtSync.getOrElse(false) && isIdeSyncing) {
+                tasks.maybeCreate("prepareKotlinIdeaImport").dependsOn(svgTask)
+            }
+
+            // Run generation before any kind of kotlin source processing
+            tasks.withType(AbstractKotlinCompile::class.java).configureEach { it.dependsOn(svgTask) }
+        }
+    }
+
+    private fun Project.estimatePackageNameOrThrow(): String {
+        val androidPluginIds = listOf(
+            "com.android.application",
+            "com.android.library",
+            "com.android.test",
+        )
+        if (androidPluginIds.any(pluginManager::hasPlugin)) {
+            return extensions
+                .findByType(CommonExtension::class.java)
+                ?.namespace
+                ?: throw GradleException(NO_PACKAGE_NAME_ERROR)
+        }
+        throw GradleException(NO_PACKAGE_NAME_ERROR)
+    }
```

Comment:
Should be defered like https://github.com/cashapp/licensee/blob/7bdc386c5a9c9c1df234c6a1e993667461e490fd/src/main/kotlin/app/cash/licensee/plugin.kt#L79-L87

Reply:
It's not called in a provider.